cli/azd/pkg/auth/manager.go

```diff
@@ -444,7 +444,11 @@ func (m *Manager) newCredentialFromCloudShell() (azcore.TokenCredential, error)
 }
 
 func (m *Manager) LoginInteractive(
-	ctx context.Context, redirectPort int, tenantID string, scopes []string) (azcore.TokenCredential, error) {
+	ctx context.Context,
+	redirectPort int,
+	tenantID string,
+	scopes []string,
+	withOpenUrl func(url string) error) (azcore.TokenCredential, error) {
```

Comment:
Nit: Since this now contains optional arguments here we may want to consider wrapping this up into `LoginOptions` struct where things like `withOpenUrl` func could be omitted when not needed.

Reply:
Updated to use options-bag